packages/cli/src/handlers/dbt/getDbtVersion.ts

```diff
@@ -24,34 +28,70 @@ export const getDbtVersion = async () => {
     }
 };
 
-export const getSupportedDbtVersion = async () => {
-    const version = await getDbtVersion();
+const getSupportedDbtVersionOption = (
+    version: string,
+): DbtVersionOption | null => {
     if (version.startsWith('1.4.')) return SupportedDbtVersions.V1_4;
     if (version.startsWith('1.5.')) return SupportedDbtVersions.V1_5;
     if (version.startsWith('1.6.')) return SupportedDbtVersions.V1_6;
     if (version.startsWith('1.7.')) return SupportedDbtVersions.V1_7;
     if (version.startsWith('1.8.')) return SupportedDbtVersions.V1_8;
     if (version.startsWith('1.9.')) return SupportedDbtVersions.V1_9;
 
-    console.error(
-        styles.warning(
-            `We don't currently support version ${version} on Lightdash, we'll be using ${DefaultSupportedDbtVersion} instead when dbt is refresh from the UI.`,
-        ),
-    );
-    return DefaultSupportedDbtVersion;
+    // No supported version found
+    return null;
+};
+
+const getFallbackDbtVersionOption = (version: string): DbtVersionOption => {
+    if (version.startsWith('1.3.')) return SupportedDbtVersions.V1_4; // legacy|deprecated support for dbt 1.3
+    return getLatestSupportDbtVersion();
 };
 
-export const isSupportedDbtVersion = (version: string) => {
-    const supportedVersions = [
-        '1.3.',
-        '1.4.',
-        '1.5.',
-        '1.6.',
-        '1.7',
-        '1.8',
-        '1.9',
-    ];
-    return supportedVersions.some((supportedVersion) =>
-        version.startsWith(supportedVersion),
-    );
+type DbtVersion = {
+    verboseVersion: string; // Verbose version returned by dbt --version
+    versionOption: DbtVersionOption; // The supported version by Lightdash
+};
+
+export const getDbtVersion = async (): Promise<DbtVersion> => {
```

Comment:
Now, we only export this function that figures out the dbt version, fallback, warning/prompt.